<launch>
  <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
    <arg name="rgb_processing" value="false" />
    <arg name="depth_registration" value="false" />
    <arg name="depth_processing" value="false" />

    <!-- We must specify an absolute topic name because if not it will be prefixed by "$(arg camera)".
         Probably is a bug in the nodelet manager: https://github.com/ros/nodelet_core/issues/7 --> 
    <arg name="scan_topic" value="/scan" />
  </include>

  <node pkg="turtlebot_calibration" type="scan_to_angle.py" name="scan_to_angle" >
    <param name="min_angle" value="-0.3" />
    <param name="max_angle" value="0.3" />
  </node>

  <node pkg="turtlebot_calibration" type="calibrate.py" name="turtlebot_calibration" output="screen" required="true">
    <remap from="imu" to="mobile_base/sensors/imu_data_raw" />
    <remap from="cmd_vel" to="mobile_base/commands/velocity" />
    <remap from="scan_angle" to="scan_angle" />
    <remap from="odom" to="odom" />
  </node>

</launch>
